### PR TITLE
Update unit test case of `Animator`

### PR DESCRIPTION
**Pull request by lixinhang**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

File: tests/index.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ function searchTests(root: string) {
     } else if (stat.isDirectory()) {
       describe(file, () => {
         searchTests(filePath);
-      });
+      }).timeout(5000);
```
Comment (luzhuang) [inline tests/index.ts:13]: 这里为啥要加个timeout?

File: tests/src/core/Animator.test.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    expect(animator["_awoken"]).to.eq(true);
+    expect(animator["_enabled"]).to.eq(true);
+    expect(animator["_onUpdateIndex"]).to.eq(0);
+    expect(animator["_phasedActive"]).to.eq(true);
+
+    // Test _tempAnimatorStateInfo default layerIndex values
+    expect(animator["_tempAnimatorStateInfo"].layerIndex).to.eq(-1);
+  });
+
+  it("animator property value", () => {
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:49]: 这个纯赋值的单测感觉没啥必要

File: tests/src/core/Animator.test.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    expect(animatorState.speed).to.eq(1);
+    expect(animatorState.wrapMode).to.eq(1);
+
+    // Test animator change play state.
+    animator.play('Walk', layerIndex, normalizedTimeOffset);
+    animatorState = animator.getCurrentAnimatorState(layerIndex);
+    expect(animatorState.name).to.eq('Walk');
+  });
+
+  it("animation enabled", () => {
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:90]: 这个纯赋值的单测感觉没啥必要

File: tests/src/core/Animator.test.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    const layerState = animatorLayerData[layerIndex].layerState;
+
+    // current animator layerState should be CrossFading(2)
+    expect(layerState).to.eq(2);
+  });
+
+  it("animation fix cross fade", () => {
+    //@ts-ignore
+    animator._reset();
+    animator.play('');
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:147]: 这个不用加

File: tests/src/core/Animator.test.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  });
+
+  it("find animator state", () => {
+    const stateName = 'Survey';
+    const expectedStateName = 'Run';
+    const layerIndex = animator["_tempAnimatorStateInfo"].layerIndex;
+
+    const currentAnimatorState = animator.getCurrentAnimatorState(layerIndex);
+    let animatorState = animator.findAnimatorState(stateName, layerIndex);
+    expect(animatorState).to.deep.eq(currentAnimatorState);
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:107]: 不用deep eq 引用相同就是一个state

File: tests/src/core/Animator.test.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+    animator.play(expectedStateName);
+    animatorState = animator.findAnimatorState(expectedStateName, layerIndex);
+    expect(animatorState).not.to.deep.eq(currentAnimatorState);
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:111]: 同上

File: tests/src/core/Animator.test.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    animator.play('Survey');
+
+    animator.enabled = false;
+    expect(animator["_enabled"]).to.eq(false);
+    animator.enabled = true;
+    expect(animator["_enabled"]).to.eq(true);
+  });
+
+  it("find animator state", () => {
+    const stateName = 'Survey';
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:101]: 少了play

File: tests/src/core/Animator.test.ts [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
...SZwdtmOlqQD2bF4AUqeiOaCJyLCH9KwDHhgDTbs9c8UwsPgKolfMnf/VXxmrSAQDAWKBL/606zybz9H+Q62k3DchNxKIApO7ISQc4QfpXAOKoPTeT3a4AFGWk4nP/8/8r3o0OkFtNbqbN4e70mUdv/M5PBwBwp5pAzjHfs5DrwzxbLP9Pc1N3P9k6mgCkkP+XpH8F4MQNNHXMDpvJzSjKL/vGfxpLOsCxDxhupvv1/tXplUf2H/u19isUvGAA4A5tEMoRR+/OeTZZFIBoZpHro03/KUWYADw36V8BWL6EYnbwld/1aCzpADkv2tFiQjLbuzK9+tjhE7/bfnnCbPeyFwwA3LGxQG5yrqOZzScAh7mezAtA0x2ddNQBOjqA9K8APNccrWm+5seeiiM6QF78nnQL/7PD5vDmbO/y9No7J0//0eSpP5hdfzzPDrxgAOiXJnBi/8/RBKArAIv9P8+eAEj/0r8C8P/pvp++EUs6wAO/EYsvFpkd1Ic3693L0+vvmlx5x/Tyn7Tpvzm8cYe+X1D6B8AGoYe6BtA0i68Ay/VRAYhFAVjG/ZTcAyD9KwBi33ug/fqzt/7Ez3Wb/g9vzHavdN+XfPWx9t7f2Y3H64Nr3VUmZ38MAOBOlIEPjbkHfvQf5Xo+AajbY7q4AaA7Oqk7dACZ9nYLQPc/1H4pkt+pddN+CdpbvuuH2/Q/vfFEu/lneu1PZzeeqPevNtOD+fJ/9scAAO6gN3zq34m5H/7v/4Oop9E0kfPJCYD7gMWS2ykAq9kBxL4evPkLPv2H/sf/st380x6zG0/W+880k735GoP0DwCr4tP/7f8q5r79yz5u0QGW6d99wALG7ReAZQfwrKybz/g3/8Pv+Nc+c3bzqXn63z36jvHI/hgAwKr54m/5xYBTLADLDiDzrZsv+l9+8Bu/4EOayc3uHqOmjiz9AwCsRwFYdgA5b9185Xe9/es+5cW5mS2X/6V/AIC1KADLDiDerZv2SxK+9k3nI0v/AABrVgCWHUCkW8OvSrAHDABgHQvAsgNIcutm+bxL/wAAw1D1lAWlNx1A+gcAWAHVamZBoa1/7gUHAFAAesuCgpoOIP0DAPSh6jMLymcD6ACeNQCAtSsAyywo8a8FzyAAgAKw7AAiIwAArEUBWHYAif+sAACA/xeb+j3hGU6HmgAAAABJRU5ErkJggg==#.glb';
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:161]: 换个url的吧  base64太长了

Comment (lixinhang) [reply tests/src/core/Animator.test.ts:161]: Url会走Request 网络请求，导致CI执行超时造成本次自动化过程失败，所以只能转成base64放置进文件进行Load

Comment (luzhuang) [reply tests/src/core/Animator.test.ts:161]: 是否能把glbResource单独放在一个文件夹比如resource之类的

File: tests/src/core/Animator.test.ts [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+canvasDOM.width = 1024;
+canvasDOM.height = 1024;
+
+describe("Animator test", function () {
+  let animator: Animator;
+  let resource: GLTFResource;
+  let engine: WebGLEngine;
+
+  before(async () => {
+    engine = await WebGLEngine.create({ canvas: canvasDOM });
```
Comment (luzhuang) [inline tests/src/core/Animator.test.ts:19]: 这里有报错，可以执行下npm run test-debug 看看

File: tests/src/core/Animator.test.ts (hunk 7 above)
Comment (luzhuang) [reply tests/src/core/Animator.test.ts:124]: it("find animator state", () => {
    const stateName = "Survey";
    const expectedStateName = "Run";
    const layerIndex = animator["_tempAnimatorStateInfo"].layerIndex;

    animator.play(stateName);
    const currentAnimatorState = animator.getCurrentAnimatorState(layerIndex);
    let animatorState = animator.findAnimatorState(stateName, layerIndex);
    expect(animatorState).to.deep.eq(currentAnimatorState);
    expect(animatorState).to.eq(currentAnimatorState);

    animator.play(expectedStateName);
    animatorState = animator.findAnimatorState(expectedStateName, layerIndex);
    expect(animatorState).not.to.eq(currentAnimatorState);
    expect(animatorState.name).to.eq(expectedStateName);
  });

**Comment by luzhuang** [comment]
有格式化问题 单引号 -> 双引号

File: tests/src/core/Animator.test.ts (hunk 3 above)
Comment (luzhuang) [reply tests/src/core/Animator.test.ts:108]: 加个update的判断就更好了：
```
it("animation enabled", () => {
    // Test animator play.
    animator.play("Survey");
    const onDisableSpy = chai.spy.on(animator, "_onDisable");
    const onEnableSpy = chai.spy.on(animator, "_onEnable");
    const onUpdateSpy = chai.spy.on(animator, "update");

    animator.enabled = false;
    expect(animator["_enabled"]).to.eq(false);
    expect(onDisableSpy).to.have.been.called.exactly(1);
    engine.update();
    expect(onUpdateSpy).to.have.been.called.exactly(0);

    animator.enabled = true;
    expect(animator["_enabled"]).to.eq(true);
    expect(onEnableSpy).to.have.been.called.exactly(1);
    engine.update();
    expect(onUpdateSpy).to.have.been.called.exactly(1);
  });
```

**Review by luzhuang — APPROVED** [review]

**Review by GuoLei1990 — APPROVED** [review]

**Pull request merged**